src/Functions/toDayOfWeek.cpp

```diff
@@ -7,14 +7,38 @@
 namespace DB
 {
 
+namespace
+{
+
+/// Sunday is 1, Monday is 2, and Saturday is 7
+struct DayOfWeekTransformer
+{
+    static constexpr auto name = "DAYOFWEEK";
+
+    static inline UInt8 transform(UInt8 src) { return src % 7 + 1; }
+};
+
+/// Monday is 0, Tuesday is 2, and Sunday is 6
+struct WeekDayTransformer
+{
+    static constexpr auto name = "WEEKDAY";
+
+    static inline UInt8 transform(UInt8 src) { return src - 1; }
+};
+
+}
+
 using FunctionToDayOfWeek = FunctionDateOrDateTimeToSomething<DataTypeUInt8, ToDayOfWeekImpl>;
+using FunctionDayOfWeek = FunctionDateOrDateTimeToSomething<DataTypeUInt8, ToDayOfWeekWithTransformImpl<DayOfWeekTransformer>>;
+using FunctionWeekDay = FunctionDateOrDateTimeToSomething<DataTypeUInt8, ToDayOfWeekWithTransformImpl<WeekDayTransformer>>;
 
 REGISTER_FUNCTION(ToDayOfWeek)
 {
     factory.registerFunction<FunctionToDayOfWeek>();
 
-    /// MysQL compatibility alias.
-    factory.registerAlias("DAYOFWEEK", "toDayOfWeek", FunctionFactory::CaseInsensitive);
```

Comment:
`DAYOFWEEK` was previously an alias for `toDayOfWeek` (Sun = 7, Mon = 1).
This is now changed such that `DAYOFWEEK` has MySQL/Spark semantics (Sun = 1, Mon = 2).

This unfortunately breaks compatibility with existing use cases/queries/scripts. One could theoretically argue that `DAYOFWEEK` behaved incorrectly for many years, but IMHO such reasoning can only apply to real bugs and not to variations of acceptable behavior. As a result, we would need to include this PR as "backward incompatible change" in the [changelog](https://clickhouse.com/docs/en/whats-new/changelog/) but I don't think this is worth the trouble.

What about this alternative (basically #40480):
- we add an optional `mode` argument to `toDayOfWeek`. Possible values are: 
    - 0: Sun = 7, Mon = 1
    - 1: Sun = 6, Mon = 0
    - 2: Sun = 0, Mon = 1
    - 3: Sun = 1, Mon = 2
- value 0 will be the default so we don't break anything.
- the documentation mentions that 3 corresponds to MySQL/Spark behavior
- alias `DAYOFWEEK` stays as is

Reply:
Ok. I'll try.